crates/mitex-parser/src/parser.rs

```diff
@@ -418,6 +434,7 @@ impl<'a, S: TokenStream<'a>> Parser<'a, S> {
                 CommandName::Left => self.item_lr(),
                 CommandName::Right => return self.command(),
                 CommandName::ErrorBeginEnvironment | CommandName::ErrorEndEnvironment => self.eat(),
+                CommandName::EndMathInline | CommandName::EndMathDisplay => return false,
```

Comment:
Eat the token and wrap an error here, see:

https://github.com/mitex-rs/mitex/blob/d2dbe88070fe6e7dc94ebb1ceb0795d710e15c0f/crates/mitex-parser/src/parser.rs#L391C13-L395

Reply:
Ooh, I see :). I couldn't figure out how to handle this